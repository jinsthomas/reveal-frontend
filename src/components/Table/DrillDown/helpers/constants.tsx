/** Constants */
import React from 'react';

export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET_SPAN = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET_SPAN = typeof CARET_SPAN;

export const DEFAULT_ROW_HEIGHT = '2em';
export type DEFAULT_ROW_HEIGHT = typeof DEFAULT_ROW_HEIGHT;
