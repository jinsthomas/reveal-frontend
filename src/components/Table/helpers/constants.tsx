/** Constants */
import React from 'react';
export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET = typeof CARET;

export const ROWS_TO_DISPLAY = 'Rows to display';
export type ROWS_TO_DISPLAY = typeof ROWS_TO_DISPLAY;

export const PAGE = 'page';
export type PAGE = typeof PAGE;

export const OF = 'of';
export type OF = typeof OF;
