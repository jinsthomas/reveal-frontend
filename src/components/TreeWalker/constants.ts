export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
