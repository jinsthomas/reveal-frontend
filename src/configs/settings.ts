/** This is the main configuration module */
import { Providers } from '@onaio/gatekeeper';
import {
  DOMAIN_NAME,
  ONADATA_ACCESS_TOKEN_URL,
  ONADATA_AUTHORIZATION_URL,
  ONADATA_CLIENT_ID,
  ONADATA_OAUTH_STATE,
  ONADATA_USER_URL,
} from './env';

/** Interfaces and Types */

/** Interface for a Focus Investigation Classification */
export interface Classification {
  code: string;
  name: string;
  description: string;
}

/** Interface that describes location items */
export interface LocationItem {
  identifier: string /** Should match the name of the column in data */;
  level: number /** The HDX-compliant level of the location in the hierarchy */;
  name: string /** The name of the location */;
}

/** Authentication Configs */
export const providers: Providers = {
  Ona: {
    accessTokenUri: ONADATA_ACCESS_TOKEN_URL,
    authorizationUri: ONADATA_AUTHORIZATION_URL,
    clientId: ONADATA_CLIENT_ID,
    redirectUri: `${DOMAIN_NAME}/oauth/callback/Ona/`,
    scopes: ['read', 'write'],
    state: ONADATA_OAUTH_STATE,
    userUri: ONADATA_USER_URL,
  },
};

/** Location configs */

/** Location item hierarchy
 * This is a list of locations.  The "level" field will be used to sort the
 * locations hierarchically, from lowest to highest.
 */
export const locationHierarchy: LocationItem[] = [
  {
    identifier: 'province',
    level: 1,
    name: 'Province',
  },
  {
    identifier: 'district',
    level: 2,
    name: 'District',
  },
  {
    identifier: 'canton',
    level: 3,
    name: 'Canton',
  },
  {
    identifier: 'village',
    level: 4,
    name: 'Village',
  },
];

/** Focus investigation configs */

/** Focus Investigation case classifications */
export const FIClassifications: Classification[] = [
  {
    code: 'A1',
    description: 'Indigenous case recorded within the last year.',
    name: 'Active',
  },
  {
    code: 'A2',
    description: 'No indigenous case during the last year, but withing the last 3 years.',
    name: 'Residual Non-Active',
  },
  {
    code: 'B1',
    description: 'Receptive area but no indigenous cases within the last 3 years.',
    name: 'Cleared Receptive',
  },
  {
    code: 'B2',
    description: 'Non-receptive area.',
    name: 'Cleared Non-Receptive',
  },
];

/** Indicators configs */

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const ONE = 0;
export const ZERO = 0;
