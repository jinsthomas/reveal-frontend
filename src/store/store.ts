import { combine } from '@onaio/redux-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { applyMiddleware, createStore, Reducer } from 'redux';
import thunk from 'redux-thunk';
import users, { reducerName as usersReducer } from './ducks/users';

export const history = createBrowserHistory();

/** Declare type for reducer Registry */
export interface Registry {
  [key: string]: Reducer;
}

/** Function to create the store */
export function getStore(reducers: Registry) {
  /** Register each of the initial reducers */
  Object.keys(reducers).forEach(reducerName => {
    reducerRegistry.register(reducerName, reducers[reducerName]);
  });
  /** Combine reducers */
  const reducer = combine(reducerRegistry.getReducers());
  /** Create the store */
  return createStore(
    reducer,
    (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk, routerMiddleware(history))
  );
}

/** Router reducer */
const connectReducer = connectRouter(history);
/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};
// /** Add users reducer to registry */
defaultReducers[usersReducer] = users;

/** The initial store for the reveal web app */
const store = getStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
  store.replaceReducer(combine(reducers));
});

export default store;
