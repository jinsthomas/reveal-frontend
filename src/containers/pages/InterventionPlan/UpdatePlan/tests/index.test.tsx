import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedUpdatePlan, { UpdatePlan } from '..';
import { PlanDefinition } from '../../../../../configs/settings';
import { PLAN_UPDATE_URL } from '../../../../../constants';
import store from '../../../../../store';
import * as fixtures from '../../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';
import { planDefinition1, planDefinition2 } from './fixtures';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

jest.mock('../PlanLocationNames', () => {
  const PlanLocationNamesMock = () => <div id="Helmuth"> No plan survives enemy contact. </div>;

  return PlanLocationNamesMock;
});

describe('components/InterventionPlan/UpdatePlan', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    fetch.resetMocks();
    jest.clearAllMocks();
  });

  function getProps() {
    const mock: any = jest.fn();
    return {
      fetchPlan: mock,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plans[1].identifier },
        path: `${PLAN_UPDATE_URL}/:id`,
        url: `${PLAN_UPDATE_URL}/${fixtures.plans[1].identifier}`,
      },
      plan: fixtures.plans[1] as PlanDefinition,
    };
  }

  it('renders without crashing', () => {
    fetch.mockResponseOnce(JSON.stringify(fixtures.plans[1]));
    shallow(
      <Router history={history}>
        <UpdatePlan {...getProps()} />
      </Router>
    );
  });

  it('renders plan definition list correctly', () => {
    fetch.mockResponseOnce(JSON.stringify(fixtures.plans[1]));
    const wrapper = mount(
      <Router history={history}>
        <UpdatePlan {...getProps()} />
      </Router>
    );
    expect(toJson(wrapper.find('Breadcrumb'))).toMatchSnapshot('Breadcrumb');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('Page title');
    expect(wrapper.find('PlanForm').props()).toMatchSnapshot({
      formHandler: expect.any(Function),
      renderLocationNames: expect.any(Function),
    });
    wrapper.unmount();
  });

  it('renders plan Location names component', () => {
    fetch.mockResponseOnce(JSON.stringify(fixtures.plans[1]));
    const wrapper = mount(
      <Router history={history}>
        <UpdatePlan {...getProps()} />
      </Router>
    );
    expect(toJson(wrapper.find('#Helmuth'))).toMatchSnapshot('Renders PlanLocation Mock');
  });

  it('pass correct data to store: API responds with array', async () => {
    // fetch with a array response
    fetch.mockResponseOnce(JSON.stringify([fixtures.plans[1]]));
    const wrapper = mount(
      <Router history={history}>
        <UpdatePlan {...getProps()} />
      </Router>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    const FetchPlanSpy = jest.spyOn(wrapper.props().children.props, 'fetchPlan');
    expect(FetchPlanSpy).toHaveBeenCalledWith(fixtures.plans[1]);
    wrapper.unmount();
  });

  it('pass correct data to store: API responds with object', async () => {
    // fetch with an object response
    fetch.mockResponseOnce(JSON.stringify(fixtures.plans[1]));
    const wrapper = mount(
      <Router history={history}>
        <UpdatePlan {...getProps()} />
      </Router>
    );
    await new Promise<unknown>(resolve => setImmediate(resolve));
    const FetchPlanSpy = jest.spyOn(wrapper.props().children.props, 'fetchPlan');
    expect(FetchPlanSpy).toHaveBeenCalledWith(fixtures.plans[1]);
  });

  it('renders case details when plan is reactive', async () => {
    fetch.once(JSON.stringify(planDefinition1));
    const props = {
      history,
      location: jest.fn(),
      match: {
        isExact: true,
        params: { id: planDefinition1.identifier },
        path: `${PLAN_UPDATE_URL}/:id`,
        url: `${PLAN_UPDATE_URL}/${planDefinition1.identifier}`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedUpdatePlan {...props} />
        </Router>
      </Provider>
    );
    // resolve promise to get plan into UpdatePlan state.
    await new Promise<unknown>(resolve => setImmediate(resolve));
    wrapper.update();

    // planDefinition1 is reactive thus we expect CaseDetails is rendered
    const caseDetailsIsRendered = wrapper.find('CaseDetails').length > 0;
    expect(caseDetailsIsRendered).toBeTruthy();
    wrapper.unmount();
  });

  it('Does not render case details when plan isnt reactive', async () => {
    fetch.once(JSON.stringify(planDefinition2));
    const props = {
      history,
      location: jest.fn(),
      match: {
        isExact: true,
        params: { id: planDefinition2.identifier },
        path: `${PLAN_UPDATE_URL}/:id`,
        url: `${PLAN_UPDATE_URL}/${planDefinition2.identifier}`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedUpdatePlan {...props} />
        </Router>
      </Provider>
    );
    // resolve promise to get plan into UpdatePlan state.
    await new Promise<unknown>(resolve => setImmediate(resolve));
    wrapper.update();

    // planDefinition2 is not reactive thus we don't expect CaseDetails to be rendered
    const caseDetailsIsntRendered = wrapper.find('CaseDetails').length === 0;
    expect(caseDetailsIsntRendered).toBeTruthy();
    wrapper.unmount();
  });
});
