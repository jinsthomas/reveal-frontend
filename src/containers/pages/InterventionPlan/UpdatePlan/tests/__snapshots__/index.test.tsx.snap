// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InterventionPlan/UpdatePlan renders plan Location names component: Renders PlanLocation Mock 1`] = `
<div
  id="Helmuth"
>
   No plan survives enemy contact. 
</div>
`;

exports[`components/InterventionPlan/UpdatePlan renders plan definition list correctly 1`] = `
Object {
  "allFormActivities": Array [
    Object {
      "actionCode": "Case Confirmation",
      "actionDescription": "Confirm the index case",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Case Confirmation",
      "goalDescription": "Confirm the index case",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 1,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "RACD Register Family",
      "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "Family Registration",
      "condition": Array [
        Object {
          "description": "Structure is residential or type does not exist",
          "expression": "$this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure'",
        },
        Object {
          "description": "Family does not exist for structure",
          "expression": "$this.contained.exists().not()",
        },
        Object {
          "description": "Apply to residential structures in Register_Structure questionnaires",
          "expression": "questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure'",
        },
      ],
      "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Register_Structure'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "RACD Register Family",
      "actionDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Family Registration",
      "goalDescription": "Register all families & family members in all residential structures enumerated (100%) within the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "Blood Screening",
      "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Blood Screening",
      "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "Blood Screening",
      "actionDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Blood Screening",
      "condition": Array [
        Object {
          "description": "Person is older than 5 years or person associated with questionnaire response if older than 5 years",
          "expression": "($this.is(FHIR.Patient) and $this.birthDate <= today() - 5 'years') or ($this.contained.where(Patient.birthDate <= today() - 5 'years').exists())",
        },
      ],
      "goalDescription": "Visit all residential structures (100%) within a 1 km radius of a confirmed index case and test each registered person",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Family Registration' or questionnaire = 'Family Member Registration'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "Bednet Distribution",
      "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Bednet Distribution",
      "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "Bednet Distribution",
      "actionDescription": "Visit 100% of residential structures in the operational area and provide nets",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "Bednet Distribution",
      "condition": Array [
        Object {
          "description": "Structure is residential or type does not exist",
          "expression": "$this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure'",
        },
        Object {
          "description": "Family exists for structure",
          "expression": "$this.contained.exists()",
        },
        Object {
          "description": "Register structure Event submitted for a residential structure",
          "expression": "questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure'",
        },
      ],
      "goalDescription": "Visit 100% of residential structures in the operational area and provide nets",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Family Registration'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "Larval Dipping",
      "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Larval Dipping",
      "condition": Array [
        Object {
          "description": "Structure is a larval breeding site",
          "expression": "$this.type.where(id='locationType').text = 'Larval Breeding Site'",
        },
        Object {
          "description": "Apply to larval breeding sites in Register_Structure questionnaires",
          "expression": "questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Larval Breeding Site'",
        },
      ],
      "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 3,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Register_Structure'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "Larval Dipping",
      "actionDescription": "Perform a minimum of three larval dipping activities in the operational area",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Larval Dipping",
      "goalDescription": "Perform a minimum of three larval dipping activities in the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 3,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "Mosquito Collection",
      "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Mosquito Collection",
      "condition": Array [
        Object {
          "description": "Structure is a mosquito collection point",
          "expression": "$this.type.where(id='locationType').text = 'Mosquito Collection Point'",
        },
        Object {
          "description": "Apply to mosquito collection point in Register_Structure questionnaires",
          "expression": "questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Mosquito Collection Point'",
        },
      ],
      "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 3,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Register_Structure'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "Mosquito Collection",
      "actionDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Mosquito Collection",
      "goalDescription": "Set a minimum of three mosquito collection traps and complete the mosquito collection process",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 3,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "IRS",
      "actionDescription": "Visit each structure in the operational area and attempt to spray",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "Spray Structures",
      "goalDescription": "Spray structures in the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 90,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "IRS",
      "actionDescription": "Visit each structure in the operational area and attempt to spray",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "Spray Structures",
      "condition": Array [
        Object {
          "description": "Structure is residential or type does not exist",
          "expression": "$this.type.where(id='locationType').exists().not() or $this.type.where(id='locationType').text = 'Residential Structure'",
        },
        Object {
          "description": "Apply to residential structures in Register_Structure questionnaires",
          "expression": "questionnaire = 'Register_Structure' and $this.item.where(linkId='structureType').answer.value ='Residential Structure'",
        },
      ],
      "goalDescription": "Spray structures in the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 90,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Register_Structure'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "MDA Dispense",
      "actionDescription": "Dispense medication to each eligible person",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "MDA Dispense",
      "goalDescription": "Dispense medication to each eligible person",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "MDA Adverse Event(s)",
      "actionDescription": "Report any adverse events from medication",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "MDA Adverse Event(s)",
      "goalDescription": "Report any adverse events from medication",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 2,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "MDA Dispense",
      "actionDescription": "Visit all residential structures (100%) and dispense prophylaxis to each registered person",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "MDA Dispense",
      "condition": Array [
        Object {
          "description": "Person is older than 5 years or person associated with questionaire response if older than 5 years",
          "expression": "($this.is(FHIR.Patient) and $this.birthDate <= today() - 5 'years') or ($this.contained.where(Patient.birthDate <= today() - 5 'years').exists())",
        },
      ],
      "goalDescription": "Visit all residential structures (100%) dispense prophylaxis to each registered person",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
        Object {
          "description": "",
          "expression": "questionnaire = 'Family Registration' or questionnaire = 'Family Member Registration'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "MDA Adherence",
      "actionDescription": "Visit all residential structures (100%) and confirm adherence of each registered person",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "MDA Adherence",
      "condition": Array [
        Object {
          "description": "The person fully received the dispense activity",
          "expression": "$this.item.where(linkId='business_status').answer.value = 'Fully Received'",
        },
      ],
      "goalDescription": "Visit all residential structures (100%) and confirm adherence of each registered person",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 100,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "description": "",
          "expression": "questionnaire = 'mda_dispense'",
          "name": "event-submission",
        },
      ],
    },
    Object {
      "actionCode": "BCC",
      "actionDescription": "Conduct BCC activity",
      "actionIdentifier": "",
      "actionReason": "Investigation",
      "actionTitle": "Behaviour Change Communication",
      "goalDescription": "Complete at least 1 BCC activity for the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 1,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
    },
    Object {
      "actionCode": "BCC",
      "actionDescription": "Conduct BCC activity",
      "actionIdentifier": "",
      "actionReason": "Routine",
      "actionTitle": "Behaviour Change Communication",
      "condition": Array [
        Object {
          "description": "Jurisdiction type location",
          "expression": "Location.physicalType.text = 'jdn'",
        },
      ],
      "goalDescription": "Complete at least 1 BCC activity for the operational area",
      "goalDue": 2017-07-20T19:31:00.000Z,
      "goalPriority": "medium-priority",
      "goalValue": 1,
      "timingPeriodEnd": 2017-07-20T19:31:00.000Z,
      "timingPeriodStart": 2017-07-13T19:31:00.000Z,
      "trigger": Array [
        Object {
          "name": "plan-activation",
        },
      ],
    },
  ],
  "allowMoreJurisdictions": true,
  "cascadingSelect": true,
  "disabledActivityFields": Array [
    "actionReason",
    "goalPriority",
  ],
  "disabledFields": Array [
    "interventionType",
    "fiReason",
    "fiStatus",
    "identifier",
    "name",
    "caseNum",
    "opensrpEventId",
    "jurisdictions",
    "activities",
    "date",
    "end",
    "fiReason",
    "fiStatus",
    "start",
    "taskGenerationStatus",
    "title",
    "version",
  ],
  "formHandler": Any<Function>,
  "initialValues": Object {
    "activities": Array [
      Object {
        "actionCode": "BCC",
        "actionDescription": "Perform BCC for the operational area",
        "actionIdentifier": "3f2eef38-38fe-4108-abb3-4e896b880302",
        "actionReason": "Routine",
        "actionTitle": "Perform BCC",
        "goalDescription": "Complete BCC for the operational area",
        "goalDue": 2019-07-10T00:00:00.000Z,
        "goalPriority": "medium-priority",
        "goalValue": 1,
        "timingPeriodEnd": 2019-07-30T00:00:00.000Z,
        "timingPeriodStart": 2019-07-10T00:00:00.000Z,
      },
      Object {
        "actionCode": "IRS",
        "actionDescription": "Visit each structure in the operational area and attempt to spray",
        "actionIdentifier": "95a5a00f-a411-4fe5-bd9c-460a856239c9",
        "actionReason": "Routine",
        "actionTitle": "Spray Structures",
        "goalDescription": "Spray 90 % of structures in the operational area",
        "goalDue": 2019-07-30T00:00:00.000Z,
        "goalPriority": "medium-priority",
        "goalValue": 90,
        "timingPeriodEnd": 2019-07-30T00:00:00.000Z,
        "timingPeriodStart": 2019-07-10T00:00:00.000Z,
      },
    ],
    "caseNum": "",
    "date": 2019-07-10T00:00:00.000Z,
    "end": 2019-07-30T00:00:00.000Z,
    "fiReason": undefined,
    "fiStatus": undefined,
    "identifier": "8fa7eb32-99d7-4b49-8332-9ecedd6d51ae",
    "interventionType": "IRS",
    "jurisdictions": Array [
      Object {
        "id": "35968df5-f335-44ae-8ae5-25804caa2d86",
        "name": "35968df5-f335-44ae-8ae5-25804caa2d86",
      },
      Object {
        "id": "3952",
        "name": "3952",
      },
      Object {
        "id": "ac7ba751-35e8-4b46-9e53-3cbaad193697",
        "name": "ac7ba751-35e8-4b46-9e53-3cbaad193697",
      },
    ],
    "name": "TwoTwoOne_01_IRS_2019-07-10",
    "opensrpEventId": undefined,
    "start": 2019-07-10T00:00:00.000Z,
    "status": "active",
    "taskGenerationStatus": "False",
    "title": "TwoTwoOne_01 IRS 2019-07-10",
    "version": "1",
  },
  "jurisdictionLabel": "Focus Area",
  "redirectAfterAction": "/plans/list",
  "renderLocationNames": Any<Function>,
}
`;

exports[`components/InterventionPlan/UpdatePlan renders plan definition list correctly: Breadcrumb 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className="reveal-breadcrumb"
  listTag="ol"
  tag="nav"
>
  <nav
    aria-label="breadcrumb"
    className="reveal-breadcrumb"
  >
    <ol
      className="breadcrumb"
    >
      <BreadcrumbItem
        key="0"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="0"
            replace={false}
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        key="1"
        tag="li"
      >
        <li
          className="breadcrumb-item"
        >
          <Link
            key="1"
            replace={false}
            to="/plans/list"
          >
            <a
              href="/plans/list"
              onClick={[Function]}
            >
              Manage Plans
            </a>
          </Link>
        </li>
      </BreadcrumbItem>
      <BreadcrumbItem
        active={true}
        tag="li"
      >
        <li
          aria-current="page"
          className="active breadcrumb-item"
        >
          Update Plan: TwoTwoOne_01 IRS 2019-07-10
        </li>
      </BreadcrumbItem>
    </ol>
  </nav>
</Breadcrumb>
`;

exports[`components/InterventionPlan/UpdatePlan renders plan definition list correctly: Page title 1`] = `
<h3
  className="mb-3 page-title"
>
  Update Plan: TwoTwoOne_01 IRS 2019-07-10
</h3>
`;
